src/parser/tiger.grm

```diff
@@ -73,3 +77,22 @@ op: PLUS ()
     | LE ()
     | GT ()
     | GE ()
+
+arraycreateexp: ID LBRACK exp RBRACK OF exp ()
+
+recordcreateexp: ID LBRACE recordcreatelistexp RBRACE ()
+
+recordcreatelistexp: recordcreatelist ()
+    | ()
```

Comment:
I wouldn't make an extra state for this. Also this should look like the code we had for `callexp` which I believe just from reading this is buggy, as it should allow `foo()` which it doesn't atm. Correct?